src/dns.c

```diff
@@ -2272,6 +2271,30 @@ hsk_dns_name_serialize(
         int j;
         for (j = begin; j < i; j++) {
           char ch = name[j];
+          if (ch == '\\') {
+            uint16_t value = 0;
+
+            // Read the next three digits and form a single byte
+            for (int x = 100; x >= 1; x /= 10) {
```

Comment:
while this is cool, the function is hard to read already 😄 maybe define an `is_digit` or use standard `isdigit` func it's only 3 checks and it'll likely be inlined so faster than loop + div op. it'll probably look more readable what do u think?